ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +20,107 @@ export interface DropDownState {
-                        return this.content && this.state.opened && !this.content.contains(event.target as Node) && !this.el.contains(event.target as Node);
+                        return content && opened && !content.contains(event.target as Node) && !el.contains(event.target as Node);
                     })
                 )
             ).subscribe(() => {
-                this.close();
+                close();
             }),
             fromEvent(document, 'scroll', {capture: true}).subscribe(() => {
-                if (this.state.opened && this.content && this.el) {
-                    this.setState(this.refreshState());
+                if (opened && content && el) {
+                    const newState = refreshState();
+
+                    setOpened(newState.opened);
+                    setLeft(newState.left);
+                    setTop(newState.top);
                 }
             })
         ];
-    }
 
-    public componentWillUnmount() {
-        (this.subscriptions || []).forEach(s => s.unsubscribe());
-        this.subscriptions = [];
-    }
+        return () => {
+            (subscriptionsRef.current || []).forEach(s => s.unsubscribe());
+            subscriptionsRef.current = [];
+        };
+    }, []);
 
-    public close() {
-        this.setState({opened: false});
-    }
-
-    private refreshState() {
-        const anchor = this.el.querySelector('.argo-dropdown__anchor') as HTMLElement;
-        const {top, left} = anchor.getBoundingClientRect();
+    const refreshState = () => {
+        const content = contentRef.current;
+        const anchor = anchorRef.current;
         const anchorHeight = anchor.offsetHeight + 2;
+        const {top: anchorTop, left: anchorLeft} = anchor.getBoundingClientRect();
+        const newState = {top: anchorTop, left: anchorLeft, opened};
 
-        const newState = {left: this.state.left, top: this.state.top, opened: this.state.opened};
         // Set top position
-        if (this.content.offsetHeight + top + anchorHeight > window.innerHeight) {
-            newState.top = top - this.content.offsetHeight - 2;
+        if (content.offsetHeight + anchorTop + anchorHeight > window.innerHeight) {
+            newState.top = anchorTop - content.offsetHeight - 2;
         } else {
-            newState.top = top + anchorHeight;
+            newState.top = anchorTop + anchorHeight;
         }
 
         // Set left position
-        if (this.content.offsetWidth + left > window.innerWidth) {
-            newState.left = left - this.content.offsetWidth + anchor.offsetWidth;
+        if (content.offsetWidth + anchorLeft > window.innerWidth) {
+            newState.left = anchorLeft - content.offsetWidth + anchor.offsetWidth;
         } else {
-            newState.left = left;
+            newState.left = anchorLeft;
         }
+
         return newState;
-    }
+    };
 
-    private open() {
-        if (!this.content || !this.el) {
+    const open = () => {
```

Comment:
```suggestion
    function open() {
```

named function same as above